Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -1340,10 +1349,50 @@ protected class ClassWriter : IClassWriter {
       TrStmt(recoveryBody, exceptBlock);
     }
 
+    private static readonly Regex ModuleLine = new(@"^\s*assert\s+""([a-zA-Z0-9_]+)""\s*==\s*__name__\s*$");
```

Comment:
Can we eliminate this trick if we require that the original file already has the right name? Java already doesn't let you name the external file `ExternN.java`.

Reply:
I think so, but I think it's convenient. It's also how it's done in go.

Reply:
The assert line was already there. The propose is to ensure the next line `module_ = sys.modules[__name__]` is actually doing the right thing. This was originally added so we could always use the full name. This could actually be in the filename, as we always flatten module hierarchies. (Imposed by the SinglePassCompiler)